src/ex4_js-objects-part2/task-11.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+function countEntrance(str) {
+  const uniqChars = [];
+  let isUnique = true;
+  for (let i = 0; i < str.length; i++) {
+    for (let j = 0; j < uniqChars.length; j++) {
+      isUnique = false;
+      if (str[i] === uniqChars[j]) break;
+      isUnique = true;
+    }
+    if (isUnique) uniqChars[uniqChars.length] = str[i];
+  }
+
+  let entCounter = 0;
```

Comment:
хороший тон все переменные объявлять в начале

Reply:
и что значит ent

Reply:
> хороший тон все переменные объявлять в начале

The style guide [13.4](https://github.com/airbnb/javascript#variables--define-where-used) might disagree since hoisting is no longer such a big deal

Reply:
> и что значит ent

Ent stands for entrance. I saw this word in the test source code for this task in "__tests__" directory. But later on I realized that "entrance" is not the proper translation of the term "вхождение" and fixed it in the fresh commit. I'll be more careful relying on other developer's source code in the future.

Reply:
in this case we have different case of using stis rule of styleguide. in their case it can call return after first checking or if no, a code continue and browser find a variable, in your case code pass without any break (i mean return)

Reply:
The example with pointless function call for assigning its result by any means at the top is one out of the many justifying this rule. But, you're right, my piece of code is probably not one of them.:grin: So, here's the commit.
BTW there's also a short [post on medium](https://medium.com/@bluepnume/theres-no-need-to-define-all-javascript-vars-once-at-the-top-of-a-function-and-there-hasn-t-been-a66b31f21822) about it.

Reply:
thanks, i'll check it later